tokio-stream/src/wrappers/lines.rs

```diff
@@ -30,6 +30,7 @@ impl<R> LinesStream<R> {
     }
 
     /// Obtain a pinned reference to the inner `Lines<R>`.
+    #[allow(clippy::wrong_self_convention)] // TODO: rename to get_pin_mut
     pub fn as_pin_mut(self: Pin<&mut Self>) -> Pin<&mut Lines<R>> {
```

Comment:
I'm not convinced these should be renamed. The clippy lint just doesn't handle this case, and allows `&mut self`. All the other methods on this type are, by virtue of `AsRef`/`AsMut`, also called `as_*`.

Reply:
In fact, there's an open issue for it. https://github.com/rust-lang/rust-clippy/issues/4546

Reply:
Good catch, that's certainly not a right clippy warning. I will replace todo with a link to the clippy issue.

(IIRC all other tokio crates name similar methods `get_pin_mut`, so I personally tend to prefer to rename it for consistency whether clippy is right or not. But that's not the topic I want to deal with in this PR...)